dialer.go

```diff
@@ -435,7 +437,22 @@ func (d *Dialer) Dial(ctx context.Context, icn string, opts ...DialOption) (conn
 	iConn := newInstrumentedConn(tlsConn, func() {
 		n := atomic.AddUint64(c.openConnsCount, ^uint64(0)) // c.openConnsCount = c.openConnsCount - 1
 		trace.RecordOpenConnections(context.Background(), int64(n), d.dialerID, cn.String())
-	}, d.dialerID, cn.String())
+	},
+		func(err error) {
```

Comment:
whitespace nit: putting this up a line would make it more obvious it's an argument at a glance.

Alternatively, you could pull this anonymous function out into a variable:

``` golang
errFunc := func(err error) {
  // ...
}
iConn := newInstrumentedConn(/*...*/, errFunc)
```